lib/mix/tasks/ecto.load.ex

```diff
@@ -51,29 +52,46 @@ defmodule Mix.Tasks.Ecto.Load do
       (typically in production)
     * `--no-compile` - does not compile applications before loading
     * `--no-deps-check` - does not check depedendencies before loading
-
+    * `--skip-if-loaded` - does not load the dump file if the repo has migrations table up
   """
 
   @impl true
-  def run(args) do
-    {opts, _} = OptionParser.parse! args, strict: @switches, aliases: @aliases
+  def run(args, table_exists? \\ &Ecto.Adapters.SQL.table_exists?/2) do
+    {opts, _} = OptionParser.parse!(args, strict: @switches, aliases: @aliases)
     opts = Keyword.merge(@default_opts, opts)
 
-    Enum.each parse_repo(args), fn repo ->
+    Enum.each(parse_repo(args), fn repo ->
       ensure_repo(repo, args)
-      ensure_implements(repo.__adapter__, Ecto.Adapter.Structure,
-                                          "load structure for #{inspect repo}")
+      ensure_implements(
+        repo.__adapter__,
+        Ecto.Adapter.Structure,
+        "load structure for #{inspect(repo)}"
+      )
+
+      {:ok, pid, _} = ensure_started(repo, all: true)
+
+      loaded? = table_exists?.(repo, "schema_migrations")
+
+      pid && repo.stop()
 
-      if skip_safety_warnings?() or
-          opts[:force] or
-          Mix.shell.yes?("Are you sure you want to load a new structure for #{inspect repo}? Any existing data in this repo may be lost.") do
-        load_structure(repo, opts)
+      cond do
+        loaded? and opts[:skip_if_loaded] ->
+          Mix.shell().info("Skipped since a structure for #{inspect(repo)} has been loaded")
```

Comment:
Since you are explicitly giving skip_if_loaded, I think we can just return `:ok`?

```suggestion
          :ok
```